packages/run-protocol/src/runStake/runStake.js

```diff
@@ -0,0 +1,171 @@
+// @ts-check
+import { handleParamGovernance } from '@agoric/governance';
+import { E, Far } from '@endo/far';
+import { makeAttestationFacets } from './attestation.js';
+import { makeRunStakeParamManager } from './params.js';
+import { makeRunStakeKit } from './runStakeKit.js';
+import { makeRunStakeManager } from './runStakeManager';
+
+/**
+ * Provide loans on the basis of staked assets that earn rewards.
+ *
+ * In addition to brands and issuers for `Staked`, `RUN`, and attestation,
+ * terms of the contract include a periodic interest rate plus
+ * a fee proportional to the amount borrowed, as well as a ratio
+ * of funds to mint and loan per unit of staked asset.
+ * These terms are subject to change by the `Electorate`
+ * and `electionManager` terms.
+ *
+ * @typedef {{
+ *   electionManager: VoteOnParamChange,
+ *   main: {
+ *     MintingRatio: ParamRecord<'ratio'>,
+ *     InterestRate: ParamRecord<'ratio'>,
+ *     LoanFee: ParamRecord<'ratio'>,
+ *     Electorate: ParamRecord<'invitation'>,
+ *   },
+ * }} RunStakeGovernanceTerms
+ *
+ * As in vaultFactory, `timerService` provides the periodic signal to
+ * charge interest according to `chargingPeriod` and `recordingPeriod`.
+ *
+ * @typedef { RunStakeGovernanceTerms & {
+ *   timerService: TimerService,
+ *   chargingPeriod: bigint,
+ *   recordingPeriod: bigint,
+ *   lienAttestationName?: string,
+ * }} RunStakeTerms
+ *
+ * The public facet provides access to invitations to get a loan
+ * or to return an attestation in order to release a lien on staked assets.
+ *
+ * @typedef {{
+ *   makeLoanInvitation: () => Promise<Invitation>,
+ *   makeReturnAttInvitation: () => Promise<Invitation>,
+ * }} RunStakePublic
+ *
+ * To take out a loan, get an `AttMaker` for your address from
```

Comment:
do we really need to abbreviate `AttestationMaker` type? clearly you'll want to name an instance shorter but the type should err towards prioritize unambiguity

Reply:
> do we really need to abbreviate `AttestationMaker` type?

No.

I inherited the name `AttMaker`. Since it was previously approved and merged, I was reluctant to change it. But I agree to change it now.